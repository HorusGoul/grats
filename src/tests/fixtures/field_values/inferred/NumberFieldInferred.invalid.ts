/** @gqlType */
export default class Query {
  /** @gqlField */
  num() {
    return 100;
  }
}
