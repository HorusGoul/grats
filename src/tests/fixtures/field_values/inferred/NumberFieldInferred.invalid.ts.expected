-----------------
INPUT
----------------- 
/** @gqlType */
export default class Query {
  /** @gqlField */
  num() {
    return 100;
  }
}

-----------------
OUTPUT
-----------------
src/tests/fixtures/field_values/inferred/NumberFieldInferred.invalid.ts:4:3 - error: Unexpected number type. GraphQL supports both Int and Float, making `number` ambiguous. Instead, import the `Int` or `Float` type from `grats` and use that. e.g. `import { Int, Float } from "grats";`.

4   num() {
    ~~~
src/tests/fixtures/field_values/inferred/NumberFieldInferred.invalid.ts:4:3 - error: Expected GraphQL field to have a type.

4   num() {
    ~~~
